cmd/aks-periscope/aks-periscope.go

```diff
@@ -22,25 +23,35 @@ func main() {
 		log.Printf("Failed to create CRD: %+v", err)
 	}
 
-	collectors := []interfaces.Collector{}
+	//create map of all possible collectors by name, discrete vars for each collector as some are diagnoser dependecies
+	allCollectorsByName := make(map[string]interfaces.Collector)
 	containerLogsCollector := collector.NewContainerLogsCollector(exporter)
-	collectors = append(collectors, containerLogsCollector)
+	allCollectorsByName["containerlogs"] = containerLogsCollector
```

Comment:
💡 Note: Lets also comment here that how this is directly relational to the `Enabled_Collectors` flag in `YAML` file, this will provide clarity and will help in understanding the new person about the `string values` and relation.

Reply:
refactored - let me know if still not clear